docs/source/user/lsp.rst

```diff
@@ -0,0 +1,100 @@
+.. Copyright (c) Jupyter Development Team.
+.. Distributed under the terms of the Modified BSD License.
+
+.. _lsp:
+
+Language Server Protocol support
+=====================================
+
+.. warning::
+
+    Language Server Protocol (LSP) support for JupyrterLab is an experimental feature. It is based on the  `jupyterlab-lsp <https://jupyterlab-lsp.readthedocs.io/>`_ extension. More detailed documentation can be found on their page.
+
+JupyterLab 4.0 ships with components on both backend and frontend to lay the infrastructure for handling and communicating with language servers.
+
+This means that other extensions can be built on top of JupyterLab's services to provide LSP features for notebooks and files.
+
+The LSP service is disabled by default, it can be enabled by togging the *Activate* option in the *Language Servers* section of the setting page.
+
+Requirements
+------------
+
+By default JupyterLab does not come with any language servers preinstalled, you need to install the servers by yourself. Supported language servers and the suggested package managers to install them are listed in the tables below:
+
+.. role:: raw-html(raw)
+    :format: html
+
+
+.. list-table:: Notebook-optimized Language Servers
```

Comment:
FYI, I am copy-editing and expanding this section in https://github.com/jupyter-lsp/jupyterlab-lsp/pull/826. Probably worth back-referencing at the end, maybe "For other community-supported language server see... (extension page)"